### ISSUE TITLE
Container cannot be created when trying to connect to a macvlan network

### ISSUE THREAD

**Issue by kevindd992002** [linuxserver/docker-plex#219]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

If you are new to Docker or this application our issue tracker is **ONLY** used for reporting bugs or requesting features. Please use [our discord server](https://discord.gg/YWrKVTn) for general support.

I'm trying to connect your Plex container to a macvlan network so that it can have its own IP in my physical L2 network. This is working fine for your Deluge container but, for some reason, is not working for your Plex container. Here's an excerpt of my docker-compose yml file:

```
version: '2.4'
networks:
  physical_network:                       
    driver: macvlan                    
    driver_opts:
      parent: enp1s0        
    ipam:
      config:
        - subnet: 192.168.20.0/24 
          gateway: 192.168.20.1    
          ip_range: 192.168.20.24/30
  deluge:
    image: linuxserver/deluge
    container_name: deluge
    depends_on:
      - pihole
    volumes:
      - /home/kevin/appdata/deluge:/config
      - /home/kevin/downloads:/downloads
      - /mnt/storage/downloads_complete:/storage/downloads_complete
    mac_address: 40:f2:d6:c7:2e:1e
    networks:
      physical_network:
        ipv4_address: 192.168.20.25
    dns:
      - 192.168.20.24
    restart: unless-stopped
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=Asia/Manila
      - UMASK_SET=002
  plex:
    image: linuxserver/plex
    container_name: plex
    depends_on:
      - pihole
    volumes:
      - /home/kevin/appdata/plex:/config
      - /mnt/storage/shows:/data/tvshows
      - /mnt/storage/movies:/data/movies
    mac_address: 2f:e6:b0:97:78:f5 
    networks:
      physical_network:
        ipv4_address: 192.168.20.26
    dns:
      - 192.168.20.24
    restart: unless-stopped
    devices:
      - /dev/dri:/dev/dri
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=Asia/Manila
      - VERSION=latest
```
------------------------------

## Expected Behavior

Container should be created with no issues.

## Current Behavior

Container is not created and errors out with this info:

```
ERROR: for plex  Cannot start service plex: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: time=\\\\\\\"2020-01-17T00:23:25+08:00\\\\\\\" level=fatal msg=\\\\\\\"failed to add interface vethca110be to sandbox: error setting interface \\\\\\\\\\\\\\\"vethca110be\\\\\\\\\\\\\\\" MAC to \\\\\\\\\\\\\\\"2f:e6:b0:97:78:f5\\\\\\\\\\\\\\\": cannot assign requested address\\\\\\\"\\\\n\\\"\"": unknown

ERROR: for plex  Cannot start service plex: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: time=\\\\\\\"2020-01-17T00:23:25+08:00\\\\\\\" level=fatal msg=\\\\\\\"failed to add interface vethca110be to sandbox: error setting interface \\\\\\\\\\\\\\\"vethca110be\\\\\\\\\\\\\\\" MAC to \\\\\\\\\\\\\\\"2f:e6:b0:97:78:f5\\\\\\\\\\\\\\\": cannot assign requested address\\\\\\\"\\\\n\\\"\"": unknown
ERROR: Encountered errors while bringing up the project.
```

## Steps to Reproduce

1. Use parameters from my yml file.
2. docker-compose up -d

## Environment
**OS:** Debian      
**CPU architecture:** x86_64
**How docker service was installed:**      

apt install curl
curl -sSL https://get.docker.com | sh

Followed docker-compose Linux installation method from website.

## Command used to create docker container (run/create/compose/screenshot)

See above.

## Docker logs

No output because container is not created. But I did some googling and came up with [this](https://github.com/moby/moby/issues/25886).

**Comment by kevindd992002** [comment]
@aptalca, do you have any ideas to this?

**Comment by thelamer** [comment]
This is not a Plex error it is a docker error, it only says Plex because of the containers name. 

Why don't you rip all that stuff out and start by adding network settings one by one to see what breaks it , you can use any container to test these settings out.

Also this is not a support platform this is for reporting bugs , which I believe this is not one.

**Comment by kevindd992002** [comment]
I was assuming it was a bug so I posted here. I know what's causing it. Plex was working fine for me when it's using the host networking mode. As soon as I changed it to connect to the macvlan network, this happens. So where do I ask for help regarding this?

**Comment by kevindd992002** [comment]
Also, like I said I already tested this with your Deluge container and it's working fine.

**Comment by kevindd992002** [comment]
If you read one of the comments in the I link I posted above, he said this:

https://github.com/moby/moby/issues/25886#issuecomment-241924453

So it's not a docker problem and is not a plex problem but an underlying-OS issue which you guys can only resolve, no?

**Comment by aptalca** [comment]
That comment is saying it's a host os problem, not container os. That's way upstream of us.

**Comment by kevindd992002** [comment]
The solution is to change the random mac address that is being assigned to the container. For some reason, docker does not want to assign the mac that I generated yesterday. I tried generating and using a new mac now and it has the same problem. 

What I did to solve it is to get the mac address of the deluge container (which I also randomly generated using the same online website), add 1 to that address and use that instead.
